### PR TITLE
feat: add bindings for BlueZ

### PR DESCRIPTION
**Pull request by acolombier**

### CODE REVIEW

File: bluez/examples/bluetoothctl.rs [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+					if !proxy.device.paired().await? {
+						continue;
+					}
+					println!(
+						"{} ({})",
+						proxy
+							.device
+							.name()
+							.await
+							.unwrap_or(proxy.device.address().await.unwrap()),
```
Comment (mmstick) [inline bluez/examples/bluetoothctl.rs:68]: Make sure not to unwrap

File: bluez/examples/bluetoothctl.rs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	// 	eprintln!("More than one adapter found. Using the first one");
+	// }
+	// let adapter = adapters.first().unwrap();
+
+	let mut parser = pico_args::Arguments::from_env();
+
+	match parser.subcommand()?.as_deref() {
+		Some("connected-devices") => match bluez_zbus::get_devices(&connection, None).await {
+			Err(why) => {
+				eprintln!("error: could not get devices: {why}");
```
Comment (mmstick) [inline bluez/examples/bluetoothctl.rs:26]: It'd be better to use `tracing::error` instead of `eprintln`

File: bluez/examples/bluetoothctl.rs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+				return Ok(ExitCode::FAILURE);
+			}
+			Ok(devices) => {
+				for proxy in devices {
+					if !proxy.device.connected().await? {
+						continue;
+					}
+					println!(
```
Comment (mmstick) [inline bluez/examples/bluetoothctl.rs:34]: tracing::debug

File: bluez/examples/bluetoothctl.rs [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+							.await
+							.unwrap_or("unknown".to_owned())
+					);
+				}
+			}
+		},
+
+		Some("paired-devices") => match bluez_zbus::get_devices(&connection, None).await {
+			Err(why) => {
+				eprintln!("error: could not get devices: {why}");
```
Comment (mmstick) [inline bluez/examples/bluetoothctl.rs:54]: tracing::error

File: bluez/examples/bluetoothctl.rs [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+						if proxy.device.paired().await? {
+							continue;
+						}
+						println!(
+							"{} ({})",
+							proxy
+								.device
+								.name()
+								.await
+								.unwrap_or(proxy.device.address().await.unwrap()),
```
Comment (mmstick) [inline bluez/examples/bluetoothctl.rs:99]: There's another unwrap here

File: bluez/examples/bluetoothctl.rs [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+								_ => None,
+							}
+						}))
+						.await
+						.into_iter()
+						.flatten()
+						.collect();
+					if devices.is_empty() {
+						eprintln!("No device found");
+						return Ok(ExitCode::FAILURE);
```
Comment (mmstick) [inline bluez/examples/bluetoothctl.rs:133]: It may be better to return a custom error type

**Review by mmstick — CHANGES_REQUESTED** [review]

**Comment by acolombier** [comment]
Hi @mmstick, thanks for your review. 
Your comments make sense for the core library, but for the example document, wouldn't it be okay to panic/use standard descriptor? Quite frankly, I spent a significant amount of time on that example already, and noticed after that other crates didn't include any example of how to use them, so if this is a blocker, I'm happy to remove the example.

**Comment by mmstick** [comment]
I guess it's OK for examples not to be perfect

**Review by mmstick — APPROVED** [review]

**Pull request merged**